components/selectors/size_of_tests.rs

```diff
@@ -14,8 +14,8 @@ use visitor::SelectorVisitor;
 size_of_test!(size_of_selector, Selector<Impl>, 8);
 size_of_test!(size_of_pseudo_element, gecko_like_types::PseudoElement, 24);
 
-size_of_test!(size_of_component, Component<Impl>, 32);
-size_of_test!(size_of_pseudo_class, PseudoClass, 24);
+size_of_test!(size_of_component, Component<Impl>, 40);
```

Comment:
Even using the `Other<Box[u16]>` enum variant type, the size of these structs is increased :(

Reply:
Hmm, that's pretty unfortunate, let's box the `Direction` then:

```rust
Dir(Box<Direction>)
```

Mentioning that it allows not growing `Component`, even though it's slightly unfortunate.

Reply:
Oh, the other thing we can do instead, and it'll fix the serialization bug I just noted below too, I think, is using `CustomIdent`, which should just be one word and allow us not to box it:

```rust
enum Direction {
    Rtl,
    Ltr,
    Other(CustomIdent),
}
```

Then you don't even need to manually write the serialization code, and you can just `#[derive(ToCss)]` on it.